hplsql/src/main/java/org/apache/hive/hplsql/Exec.java

```diff
@@ -1044,6 +1048,12 @@ public void printExceptions() {
         trace(null, "Signal: " + sig.type);
       }
     }
+    // if there are any user defined signals then push them back to signals stack to handle them later.
+    if (userDefinedSignals.size() > 0) {
+      for (int i = userDefinedSignals.size() - 1; i >= 0; i--) {
```

Comment:
Is this check necessary? If `userDefinedSignals` is empty the loop doesn't executed.

Reply:
Fixed